nebu/cli/_common.py

```diff
@@ -7,6 +7,10 @@
 from ..logger import configure_logging, logger
 from .exceptions import UnknownEnvironment
 
+# Domains where the archive endpoint is a subdomain of the environment URL
+ARCHIVE_AS_SUBDOMAIN = [
+    "local.cnx.org"
```

Comment:
is there a disadvantage to just using archive-local.cnx.org and not requiring this special case?

Reply:
I thought about that. It would be a change to the ansible configuration, though I'm not sure what else out there may be wired to expect `archive-local.cnx.org` as that variable has been the same for 5 years. It also just seemed odd to change domain names there due to a tool limitation. At the same time, it's not that I'm particularly happy about this change either (explicit exception list is very ugly). 

In general, it seems to be an artifact of hard coding a convention versus having it be an explicit configuration. The latter would be the best alternative IMO, but I didn't go too far down that path as I suspect there are many places that would be impacted by making a change like that.

Reply:
Actually, perhaps there's a hybrid: We can add support for parsing an `archive-url` in the .ini file, and if it's not there then fallback to the convention as a backwards compatibility. 

Thoughts?

Reply:
You mean to specify via config what the archive url will look like? Yeah, I like the idea of that!

Reply:
Yes. So, for example the following would address the problem scenario:

```
[environ-vagrant]
url = https://local.cnx.org
archive_url = https://archive.local.cnx.org
```

But the this would still work and the code would construct the `archive_url` the way it does today:

```
[environ-vagrant]
url = https://staging.cnx.org
```

I'll spin that up real quick.

Reply:
Would an environment variable (or CLI parameter) remove the need for a configuration file without complicating the code too much? If so, that could simplify the `fetch-book` script a little as well as the vagrant instructions.

Reply:
Just to clarify @philschatz  , is your suggestion to make a change to `neb` where it can use command line / environment variables in lieu of a configuration file altogether? (e.g. not specific to the `archive_url` parameter, but in general for both URLs)

Reply:
Yes, that is what I was suggesting. I think the config file is for the content-manager use-case (shorthand names for different environments) but is not as useful for VM dev or where the server is dynamically provided

Reply:
Got it, and I can see your point on the use case variance based upon personas. However, I do think it's somewhat orthogonal to this specific issue which pertains to not having neb construct an incorrect `archive_url` based on using a convention. I think we should support being able to specify that in the `.ini` file. Separately, I think having CLI arguments / environment variables to support the absence of a `.ini` file is worth pursuing, but it seems like a separate issue (to me at least). Happy to create and tackle that separate issue (e.g. this isn't me punting on it =) ).

FWIW, as a user I'd probably still want the `.ini` file in my environment configured because I'd have to type less and not worry about forgetting to update environment variables when switching between qa / local.cnx / ... . For automation, it does seem like args / env vars would be nice for sure.

Reply:
@tomjw64 , @philschatz - I've updated the PR with a change to consume `archive_url` from the `.ini` file if available and otherwise fallback to the existing construction logic.

Separately, I think supporting environment variables could be done with some changes to `discover_settings` [here](https://github.com/openstax/nebuchadnezzar/blob/1583b87ba059f677e37692659551fcc3e9976b12/nebu/config.py#L67). There are a few questions that come to mind as part of doing that (such as precedence if both configuration options are found to exist), but if you agree adding this support should be a separate issue I can create that and add those there.